SignalServiceKit/src/Messages/OWSBatchMessageProcessor.m

```diff
@@ -263,16 +278,27 @@ - (void)databaseViewRegistrationComplete
 
 #pragma mark - instance methods
 
+- (dispatch_queue_t)serialQueue
+{
+    static dispatch_queue_t queue = nil;
```

Comment:
It turns out that using dedicated serial queues in the message decrypter and message batch processor queues is necessary.  It allows us to skip a `dispatch_async()` when processing many messages and takes load off the main thread.